### ISSUE TITLE
Seal `HasCompact` + `CompactLen`

### ISSUE THREAD

**Issue by bkchr** [paritytech/parity-scale-codec#514]
Both traits should probably not be implemented by anyone downstream. We should check again, but the can be very likely be sealed.

**Comment by bkchr** [comment]
We should check the usage in `ruint` and then decide on what to do. Maybe `HasCompact` could also gets a better name :see_no_evil: `SupportsCompact`? Or whatever. Maybe the entire trait structure is to complicated there and requires some rethinking.

**Comment by bkchr** [comment]
I mean we already had some breaking changes ;) Otherwise we would not have ended at 3.x. If you want to redesign it, do it :P

**Comment by koute** [comment]
Yeah, I know. :P I don't know if I have the strength to do the redesign *right now* (but at least with the monorepo in place for Polkadot updating everything would probably not make me want to immediately want to gouge my eyes out), just wanted to voice my three cents. I'm not going to block 4.0 as-is or anything like that. :P (If the only breaking change is the one we had then the changes required downstream should be relatively minimal anyway.)
